### PR TITLE
[DOC] Fix code / doc mismatches in regards to `activation` for `TapNetRegressor`, `CNTCNetwork`, and `MACNNRegressor`

### LINKED ISSUE #8890: [DOC] Fix the documentation in the context of activation for `TapNetRegressor`, `CNTCNetwork` and `MACNNRegressor`
Documentation for `TapNetRegressor`, `CNTCNetwork` and `MACNNRegressor` is a bit off in regards to `activation` parameter.

(Code mismatches documentation)

More context can be found in this post: https://github.com/sktime/sktime/pull/8880#issuecomment-3365453018

### PR DESCRIPTION
**Pull request by noxthot**
#### Reference Issues/PRs
Fixes #8890

#### What does this implement/fix? Explain your changes.
`TapNetRegressor`:
- Change `activation` default from `sigmoid` to `linear` in docstring (since the code previously defaulted to `None` which is mapped to `linear` by `keras`).
- Default `activation` to `"linear"` instead of `None` for clarity. ([`keras` documentation](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Dense) states that `If you don't specify anything, no activation is applied (ie. "linear" activation: a(x) = x).) -> No functional change.

`CNTCNetwork`:
- Remove `activation` from docstring since it was never exposed.

`MACNNRegressor`:
- Remove `activation` from function header since it is neither documented nor used.

#### Does your contribution introduce a new dependency? If yes, which one?
No

#### Did you add any tests for the change?
No

#### Any other comments?
No

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/sktime/blob/main/.all-contributorsrc) in the `sktime` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://github.com/all-contributors/all-contributors/blob/master/docs/emoji-key.md)
- [ ] Optionally, for added estimators: I've added myself and possibly to the `maintainers` tag - do this if you want to become the owner or maintainer of an estimator you added.
  See here for further details on the [algorithm maintainer role](https://www.sktime.net/en/latest/get_involved/governance.html#algorithm-maintainers).
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

##### For new estimators
- [ ] I've added the estimator to the API reference - in `docs/source/api_reference/taskname.rst`, follow the pattern.
- [ ] I've added one or more illustrative usage examples to the docstring, in a pydocstyle compliant `Examples` section.
- [ ] If the estimator relies on a soft dependency, I've set the `python_dependencies` tag and ensured
  dependency isolation, see the [estimator dependencies guide](https://www.sktime.net/en/latest/developer_guide/dependencies.html#adding-a-soft-dependency).

### CODE REVIEW

File: sktime/regression/deep_learning/tapnet.py [hunk 1]
```diff
@@ -92,7 +92,7 @@ def __init__(
         dilation=1,
         layers=(500, 300),
         use_rp=True,
-        activation=None,
+        activation="linear",
```
Comment (fkiraly) [inline sktime/regression/deep_learning/tapnet.py:95]: is this changing the default or just mkaing it explicit?

**Review by fkiraly — CHANGES_REQUESTED** [review]
short question on the default of `activation` in tapnet

File: sktime/regression/deep_learning/tapnet.py (hunk 1 above)
Comment (noxthot) [reply sktime/regression/deep_learning/tapnet.py:95]: Yes; it is just making it explicit. I unfortunately messed up the first post (commented most of the text accidentally); i have fixed it now.

This is the relevant part:
> Default activation to "linear" instead of None for clarity. ([keras documentation](https://www.tensorflow.org/api_docs/python/tf/keras/layers/Dense) states that `If you don't specify anything, no activation is applied (ie. "linear" activation: a(x) = x).) -> No function change.

**Comment by noxthot** [comment]
> short question on the default of `activation` in tapnet

answered here: https://github.com/sktime/sktime/pull/8899#discussion_r2404353977

And in the description of the pull request.

**Review by fkiraly — APPROVED** [review]
👍
